src/crypto/sign.js

```diff
@@ -0,0 +1,298 @@
+		publicKeyHeader,
+		publicKey,
+		signatureHeader,
+		signedMessage,
+		signatureFooter,
+	];
+
+	return outputArray.join('\n');
+}
+
+/**
+ * @method signAndPrintMessage
+ * @param message
+ * @param secret
+ *
+ * @return {string}
+ */
+
+export function signAndPrintMessage(message, secret) {
+	const { publicKey } = getPrivateAndPublicKeyFromSecret(secret);
+	const signedMessage = signMessageWithSecret(message, secret);
+
+	return printSignedMessage(message, signedMessage, publicKey);
+}
+
+/**
+ * @method encryptMessageWithSecret
+ * @param message
+ * @param secret
+ * @param recipientPublicKey
+ *
+ * @return {object}
+ */
+
+export function encryptMessageWithSecret(message, secret, recipientPublicKey) {
+	const senderPrivateKeyBytes = getRawPrivateAndPublicKeyFromSecret(secret).privateKey;
+	const convertedPrivateKey = convertPrivateKeyEd2Curve(senderPrivateKeyBytes);
+	const recipientPublicKeyBytes = hexToBuffer(recipientPublicKey);
+	const convertedPublicKey = convertPublicKeyEd2Curve(recipientPublicKeyBytes);
+	const messageInBytes = naclInstance.encode_utf8(message);
+
+	const nonce = naclInstance.crypto_box_random_nonce();
+	const cipherBytes = naclInstance.crypto_box(
+		messageInBytes, nonce, convertedPublicKey, convertedPrivateKey,
+	);
+
+	const nonceHex = bufferToHex(nonce);
+	const encryptedMessage = bufferToHex(cipherBytes);
+
+	return {
+		nonce: nonceHex,
+		encryptedMessage,
+	};
+}
+
+/**
+ * @method decryptMessageWithSecret
+ * @param cipherHex
+ * @param nonce
+ * @param secret
+ * @param senderPublicKey
+ *
+ * @return {string}
+ */
+
+export function decryptMessageWithSecret(cipherHex, nonce, secret, senderPublicKey) {
+	const recipientPrivateKeyBytes = getRawPrivateAndPublicKeyFromSecret(secret).privateKey;
+	const convertedPrivateKey = convertPrivateKeyEd2Curve(recipientPrivateKeyBytes);
+	const senderPublicKeyBytes = hexToBuffer(senderPublicKey);
+	const convertedPublicKey = convertPublicKeyEd2Curve(senderPublicKeyBytes);
+	const cipherBytes = hexToBuffer(cipherHex);
+	const nonceBytes = hexToBuffer(nonce);
+
+	const decoded = naclInstance.crypto_box_open(
+		cipherBytes, nonceBytes, convertedPublicKey, convertedPrivateKey,
+	);
+
+	return naclInstance.decode_utf8(decoded);
+}
+
+/**
+ * @method signTransaction
+ * @param transaction Object
+ * @param givenKeys Object
+ *
+ * @return {string}
+ */
+
+export function signTransaction(transaction, givenKeys) {
+	const transactionHash = getTransactionHash(transaction);
+	const signature = naclInstance.crypto_sign_detached(transactionHash, Buffer.from(givenKeys.privateKey, 'hex'));
+	return Buffer.from(signature).toString('hex');
+}
+
+/**
+ * @method multiSignTransaction
+ * @param transaction Object
+ * @param givenKeys Object
+ *
+ * @return {string}
+ */
+
+export function multiSignTransaction(transaction, givenKeys) {
+	const transactionToSign = Object.assign({}, transaction);
+	delete transactionToSign.signature;
+	delete transactionToSign.signSignature;
```

Comment:
Not for this PR, but just pointing out if we included the babel plugin for object destructuring we could do this:
```js
const {
    signature,
    signSignature,
    ...transactionToSign,
} = transaction;
```